openverse_api/test/dead_link_filter_test.py

```diff
@@ -0,0 +1,137 @@
+from test.constants import API_URL
+from unittest.mock import MagicMock, patch
+
+import pytest
+import requests
+
+
+def _patch_redis():
+    def redis_mget(keys, *_, **__):
+        """
+        Patch for ``redis.mget`` used by ``validate_images`` to use validity
+        information from the cache
+        """
+        return [None] * len(keys)
+
+    mock_conn = MagicMock()
+    mock_conn.mget = MagicMock(side_effect=redis_mget)
+    return patch("django_redis.get_redis_connection", return_value=mock_conn)
+
+
+def _path_grequests():
+    def grequests_map(reqs, *_, **__):
+        """
+        Patch for ``grequests.map`` used by ``validate_images`` to filter
+        and remove dead links
+        """
+        responses = []
+        for idx in range(len(list(reqs))):
+            mocked_res = MagicMock()
+            mocked_res.status_code = 200 if idx % 10 != 0 else 404
+            responses.append(mocked_res)
+        return responses
+
+    return patch("grequests.map", side_effect=grequests_map)
+
+
+@pytest.mark.django_db
+def test_dead_link_filtering(client):
+    _patch_redis().start()
+    mocked_map = _path_grequests().start()
```

Comment:
Why use the patches this way? If we do, we need to call `stop` on them: https://docs.python.org/3/library/unittest.mock.html#patch-methods-start-and-stop

Using the decorator syntax might be preferrable though:

```suggestion
@pytest.mark.django_db
@_patch_redis()
@_path_grequests()
def test_dead_link_filtering(mocked_map, mocked_redis, client):
```

I think something like that would work? If not then we'll need to carefully call `.stop` on the patch result and make sure that it is called no matter where the test fails (as each assertion wil cause the method to break out and `stop` at the end wouldn't get called)

Reply:
The decorator syntax works great, thanks for recommending it! Implemented the change in 49373fd.